src/panels/CreateFleetPanel.ts

```diff
@@ -0,0 +1,136 @@
+    private readonly resourceManagementClient: ResourceManagementClient;
+    private readonly fleetClient: ContainerServiceFleetClient;
+    private readonly featureClient: FeatureClient;
+    // private readonly commandId: string;
+
+    constructor(
+        private readonly sessionProvider: ReadyAzureSessionProvider,
+        private readonly subscriptionId: string,
+        private readonly subscriptionName: string,
+        // commandId: string,
+    ) {
+        this.resourceManagementClient = getResourceManagementClient(sessionProvider, this.subscriptionId);
+        this.fleetClient = getAksFleetClient(sessionProvider, this.subscriptionId);
+        this.featureClient = getFeatureClient(sessionProvider, this.subscriptionId);
+        // this.commandId = commandId;
+        console.log(this.resourceManagementClient, this.featureClient, this.sessionProvider);
+    }
+
+    getTitle(): string {
+        return "Create Fleet";
+    }
+
+    getInitialState(): InitialState {
+        return {
+            subscriptionId: this.subscriptionId,
+            subscriptionName: this.subscriptionName,
+        };
+    }
+
+    getTelemetryDefinition(): TelemetryDefinition<"createFleet"> {
+        return {
+            getResourceGroupsRequest: false,
+            getLocationsRequest: false,
+            createFleetRequest: true,
+        };
+    }
+
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            getLocationsRequest: () => this.handleGetLocationsRequest(webview),
+            getResourceGroupsRequest: () => this.handleGetResourceGroupsRequest(webview),
+            createFleetRequest: (args) =>
+                this.handleCreateFleetRequest(
+                    args.isNewResourceGroup,
+                    args.resourceGroupName,
+                    args.location,
+                    args.name,
+                ),
+            // args.isNewResourceGroup,
+            // args.resourceGroupName,
+            // args.location,
+            // args.name,
+            // args.hubClusterMode,
+            // webview,
+        };
+    }
+
+    private handleGetLocationsRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        console.log(webview);
+    }
+
+    private handleGetResourceGroupsRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        console.log(webview);
+    }
+
+    private async handleCreateFleetRequest(
+        isNewResourceGroup: boolean,
+        resourceGroupName: string,
+        location: string,
+        name: string,
+        // hubClusterMode: HubClusterMode,
+        // webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        if (isNewResourceGroup) {
+            // create new resource group
+        }
+
+        const resource = {
+            location: location,
+        };
+
+        try {
+            await createFleet(this.fleetClient, resourceGroupName, name, resource);
+        } catch (error) {
+            console.log(error);
```

Comment:
Please throw the error message as notification to the user. Please refer other CreateClusterPanel.ts for error handling.